Test failure in CI build 8109

The following test appears to have failed:

[#8109](https://circleci.com/gh/cockroachdb/cockroach/8109):

```

```
Please assign, take a look and update the issue accordingly.

@maxlang @tamird 

```
touch bower.installed
node_modules/.bin/tsd install

-> running install
[ERR!] GitHub API: GitHub rate limit reached.
    To increase the limit use GitHub authentication.
    See: https://github.com/DefinitelyTyped/tsd#tsdrc

Makefile:62: recipe for target 'tsd.installed' failed
make: *** [tsd.installed] Error 1
ui/ui.go:22: running "make": exit status 2
lxc-start: lxc_start.c: main: 342 The container failed to start.
```

--- a/build/Dockerfile
+++ b/build/Dockerfile
@@ -1,0 +1,16 @@
+FROM golang:1.5
+
+MAINTAINER Peter Mattis <peter@cockroachlabs.com>
+
+RUN curl --silent --location https://deb.nodesource.com/setup_4.x | bash - && \
+ apt-get dist-upgrade -y && \
+ apt-get install --no-install-recommends --auto-remove -y git build-essential file nodejs && \
+ apt-get clean autoclean && \
+ apt-get autoremove -y && \
+ rm -rf /tmp/*
+
+RUN go get golang.org/x/tools/cmd/vet
+
+ENV SKIP_BOOTSTRAP=1
+
+CMD ["/bin/bash"]

--- a/build/build-docker-deploy.sh
+++ b/build/build-docker-deploy.sh
@@ -27,4 +27,4 @@ fi
 $(dirname $0)/builder.sh $0 docker
 
 # Build the image.
-docker build -t cockroachdb/cockroach "$(dirname $0)/deploy"
+docker build --tag=cockroachdb/cockroach "$(dirname $0)/deploy"

--- a/build/builder.sh
+++ b/build/builder.sh
@@ -5,20 +5,7 @@ set -eu
 image="cockroachdb/builder"
 
 function init() {
-  docker build --tag="${image}" - <<EOF
-FROM golang:1.5
-
-RUN curl --silent --location https://deb.nodesource.com/setup_4.x | bash - && \
- apt-get dist-upgrade -y && \
- apt-get install --no-install-recommends --auto-remove -y git build-essential file nodejs && \
- apt-get clean autoclean && \
- apt-get autoremove -y && \
- rm -rf /tmp/*
-RUN go get golang.org/x/tools/cmd/vet
-ENV SKIP_BOOTSTRAP=1
-
-CMD ["/bin/bash"]
-EOF
+  docker build --tag="${image}" "$(dirname $0)"
 }
 
 if [ "${1-}" = "pull" ]; then

--- a/build/builder.sh
+++ b/build/builder.sh
@@ -41,19 +41,20 @@ if [ -t 0 ]; then
   tty="--tty"
 fi
 
+buildcache_dir="buildcache"
+uicache_dir="uicache"
+
 # Run our build container with a set of volumes mounted that will
 # allow the container to store persistent build data on the host
 # computer.
 # -i causes some commands (including `git diff`) to attempt to use
 # a pager, so we override $PAGER to disable.
 docker run -i ${tty-} ${rm} \
-  --volume="${gopath0}/src:/go/src" \
+  --volume="${gopath0}:/go" \
+  --volume="${HOME}/${buildcache_dir}:/${buildcache_dir}" \
+  --volume="${HOME}/${uicache_dir}:/${uicache_dir}" \
   --volume="${PWD}:/go/src/github.com/cockroachdb/cockroach" \
-  --volume="${gopath0}/pkg:/go/pkg" \
-  --volume="${gopath0}/pkg/linux_amd64_netgo:/usr/src/go/pkg/linux_amd64_netgo" \
-  --volume="${gopath0}/pkg/linux_amd64_race:/usr/src/go/pkg/linux_amd64_race" \
-  --volume="${gopath0}/bin/linux_amd64:/go/bin" \
   --workdir="/go/src/github.com/cockroachdb/cockroach" \
-  --env="CACHE=/go/pkg/cache" \
+  --env="CACHE=/${buildcache_dir}" \
   --env="PAGER=cat" \
   "${image}" "$@"

--- a/build/circle-deps.sh
+++ b/build/circle-deps.sh
@@ -11,12 +11,14 @@ set -eux
 if [ "${1-}" = "docker" ]; then
   cmds=$(grep '^cmd' GLOCKFILE | grep -v glock | awk '{print $2}')
 
-  # `"$GOPATH"/pkg/cache` is `cachedir` on the host computer.
+  # `/uicache` is `~/uicache` on the host computer, which is cached by
+  # circle-ci.
+
   # Create the cache directories to avoid errors on `ln` below.
-  mkdir -p "$GOPATH"/pkg/cache/{bower_components,node_modules,typings}
+  mkdir -p /uicache/{bower_components,node_modules,typings}
 
   # Symlink the cache into the source tree.
-  ln -s "$GOPATH"/pkg/cache/{bower_components,node_modules,typings} ui/
+  ln -s /uicache/{bower_components,node_modules,typings} ui/
   time make -C ui {bower,npm,tsd}.installed
 
   # Restore previously cached build artifacts.
@@ -33,23 +35,23 @@ if [ "${1-}" = "docker" ]; then
   exit 0
 fi
 
-gopath0="${GOPATH%%:*}"
-cachedir="${gopath0}/pkg/cache"
+# `~/buildcache` is cached by circle-ci.
+buildcache_dir=~/buildcache
 
 # The tag for the cockroachdb/builder image. If the image is changed
 # (for example, adding "npm"), a new image should be pushed using
 # "build/builder.sh push" and the new tag value placed here.
 tag="20150929-133252"
 
-mkdir -p "${cachedir}"
-du -sh "${cachedir}"
-ls -lh "${cachedir}"
+mkdir -p "${buildcache_dir}"
+du -sh "${buildcache_dir}"
+ls -lh "${buildcache_dir}"
 
 if ! docker images | grep -q "${tag}"; then
   # If there's a base image cached, load it. A click on CircleCI's "Clear
   # Cache" will make sure we start with a clean slate.
-  rm -f "$(ls ${cachedir}/builder*.tar 2>/dev/null | grep -v ${tag})"
-  builder="${cachedir}/builder.${tag}.tar"
+  rm -f "$(ls ${buildcache_dir}/builder*.tar 2>/dev/null | grep -v ${tag})"
+  builder="${buildcache_dir}/builder.${tag}.tar"
   if [[ ! -e "${builder}" ]]; then
     time docker pull "cockroachdb/builder:${tag}"
     docker tag -f "cockroachdb/builder:${tag}" "cockroachdb/builder:latest"
@@ -71,6 +73,6 @@ $(dirname $0)/builder.sh $0 docker
 # Clear the cache of any files that haven't been modified in more than
 # 12 hours. Note that the "build-cache restore" call above will
 # "touch" any cache files that are used by the current run.
-find "${cachedir}" -type f -mmin +720 -ls -delete
-du -sh "${cachedir}"
-ls -lh "${cachedir}"
+find "${buildcache_dir}" -type f -mmin +720 -ls -delete
+du -sh "${buildcache_dir}"
+ls -lh "${buildcache_dir}"

--- a/build/circle-deps.sh
+++ b/build/circle-deps.sh
@@ -11,6 +11,14 @@ set -eux
 if [ "${1-}" = "docker" ]; then
   cmds=$(grep '^cmd' GLOCKFILE | grep -v glock | awk '{print $2}')
 
+  # `"$GOPATH"/pkg/cache` is `cachedir` on the host computer.
+  # Create the cache directories to avoid errors on `ln` below.
+  mkdir -p "$GOPATH"/pkg/cache/{bower_components,node_modules,typings}
+
+  # Symlink the cache into the source tree.
+  ln -s "$GOPATH"/pkg/cache/{bower_components,node_modules,typings} ui/
+  time make -C ui {bower,npm,tsd}.installed
+
   # Restore previously cached build artifacts.
   time go install github.com/cockroachdb/build-cache
   time build-cache restore . .:race,test ${cmds}
@@ -31,7 +39,7 @@ cachedir="${gopath0}/pkg/cache"
 # The tag for the cockroachdb/builder image. If the image is changed
 # (for example, adding "npm"), a new image should be pushed using
 # "build/builder.sh push" and the new tag value placed here.
-tag="20150928-162648"
+tag="20150929-133252"
 
 mkdir -p "${cachedir}"
 du -sh "${cachedir}"

--- a/circle.yml
+++ b/circle.yml
@@ -10,6 +10,9 @@ checkout:
 dependencies:
   override:
     - build/circle-deps.sh
+  cache_directories:
+    - ~/buildcache
+    - ~/uicache
 
 test:
   override:

